src/transformStateWithClones.js

```diff
@@ -65,7 +65,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const copy = { ...state };
+  const result = [];
+
+  for (const { operation, properties } of transforms) {
+    if (operation === 'addProperties') {
+      for (const key in properties) {
+        copy[key] = properties[key];
+      }
+
+      result.push({ ...copy });
```

Comment:
you can push a copied object into results, outside of the loop, in that case, you write this line only once instead of three